vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -608,9 +613,15 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
                     _self._properties.context.telemetryTrace.traceID = Util.newId();
                     _self._properties.context.telemetryTrace.name = window.location.pathname;
                 }
+                if (this._currUri) {
+                    this._prevUri = this._currUri;
+                    this._currUri = window.location.href;
+                } else {
+                    this._currUri = window.location.href;
+                }
                 setTimeout(() => {
                     // todo: override start time so that it is not affected by autoRoutePVDelay
-                    _self.trackPageView({ refUri: null, properties: { duration: 0 } }); // SPA route change loading durations are undefined, so send 0
+                    _self.trackPageView({ refUri: this._prevUri, properties: { duration: 0 } }); // SPA route change loading durations are undefined, so send 0
```

Comment:
This still won't work since the stuff inside `setTimeout` will hoist out the most up to date version of `this._prevUri`, which means it will still miss fast changing URLs. Instead you can try
```ts
setTimeout(((uri: string) => {
  // todo: override start time so that it is not affected by autoRoutePVDelay
  _self.trackPageView({ refUri: uri, properties: { duration: 0 } }); // SPA route change loading durations are undefined, so send 0
}).bind(this, this._prevUri), _self.autoRoutePVDelay);
```